creusot/src/callbacks.rs

```diff
@@ -23,6 +43,27 @@ impl Callbacks for ToWhy {
                 cleanup_spec_closures(tcx, def_id.def_id_for_type_of(), &mut mir);
                 tcx.alloc_steal_mir(mir)
             };
+
+            providers.mir_borrowck = |tcx, def_id| {
+                // We use `mir_promoted` as it is the MIR required by borrowck
+                let (body, promoted) = tcx.mir_promoted(ty::WithOptConstParam::unknown(def_id));
+                let body_and_proms = BodyAndPromoteds {
+                    body: body.borrow().clone(),
+                    promoted: promoted.borrow().clone(),
+                };
+
+                // SAFETY: The reader casts the 'static lifetime to 'tcx before using it.
+                let body_and_proms: BodyAndPromoteds<'static> =
+                    unsafe { std::mem::transmute(body_and_proms) };
```

Comment:
uh. is there no better way to do this? (I can't think of any off the top of my head)

Reply:
it's kind of stupid but I don't see any other way since storage in the thread local requires `'static`